### PR TITLE
Swap metrics tests

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Refactoring and tests for sale related modules. The store was not split because the solution could be replaced w/ a canister soon.

# Changes

- move querySnsSwapMetrics to api
- move parseSnsSwapSaleBuyerCount to utils

# Tests

- api/querySnsSwapMetrics
- services/loadSnsSwapMetrics
- utils/parseSnsSwapSaleBuyerCount
- snsSwapMetricsStore
- display total participants in ProjectCommitment

### CODE REVIEW

File: frontend/src/tests/lib/components/project-detail/ProjectCommitment.spec.ts [hunk 1]
```diff
@@ -30,6 +31,30 @@ describe("ProjectCommitment", () => {
     ).toEqual(`${formatToken({ value: summary.swap.params.min_icp_e8s })} ICP`);
   });
 
+  it("should render total participants", () => {
+    snsSwapMetricsStore.setMetrics({
+      rootCanisterId: mockSnsFullProject.swapCommitment.rootCanisterId,
+      metrics: {
+        saleBuyerCount: 666,
```
Comment (lmuntaner) [inline frontend/src/tests/lib/components/project-detail/ProjectCommitment.spec.ts:38]: Can you move `666` to a constant and use it in the `expect` afterwards?

File: frontend/src/tests/lib/services/sns-swap-metrics.services.spec.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import { loadSnsSwapMetrics } from "$lib/services/sns-swap-metrics.services";
+import { snsSwapMetricsStore } from "$lib/stores/sns-swap-metrics.store";
+import { mockPrincipal } from "$tests/mocks/auth.store.mock";
+import { Principal } from "@dfinity/principal";
+import { get } from "svelte/store";
+
+describe("sns-swap-metrics", () => {
+  const RAW_METRICS = `
+# TYPE sale_buyer_count gauge
+sale_buyer_count 33 1677707139456
```
Comment (lmuntaner) [inline frontend/src/tests/lib/services/sns-swap-metrics.services.spec.ts:15]: Can you add the `33` as a variable and use it in the tests?

File: frontend/src/tests/lib/services/sns-swap-metrics.services.spec.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        .spyOn(snsSwapMetrics, "querySnsSwapMetrics")
+        .mockResolvedValue(RAW_METRICS);
+      await loadSnsSwapMetrics({
+        rootCanisterId,
+        swapCanisterId,
+        forceFetch: true,
+      });
+
+      expect(querySnsSwapMetricsSpy).toBeCalledTimes(1);
+      expect(
```
Comment (lmuntaner) [inline frontend/src/tests/lib/services/sns-swap-metrics.services.spec.ts:79]: Could you add an `expect` before this one that the value is not 33?

File: frontend/src/tests/lib/utils/sns.utils.spec.ts [hunk 4]
```diff
@@ -375,4 +376,34 @@ describe("sns-utils", () => {
       ).toBeFalsy();
     });
   });
+
+  describe("parseSnsSwapSaleBuyerCount", () => {
+    const RAW_METRICS = `
+# TYPE sale_buyer_count gauge
+sale_buyer_count 33 1677707139456
```
Comment (lmuntaner) [inline frontend/src/tests/lib/utils/sns.utils.spec.ts:383]: Ditto with "33" in a variable

**Review by lmuntaner — COMMENTED** [review]
Just some minor changes, already approving.

Thanks!

File: frontend/src/lib/utils/sns.utils.ts [hunk 5]
```diff
@@ -258,3 +258,25 @@ export const hasOpenTicketInProcess = ({
   // `undefined` means that we could still be polling for the ticket.
   return { status: projectTicketData.keepPolling ? "polling" : "none" };
 };
+
+/**
+ * Exported for testing purposes
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/utils/sns.utils.ts:263]: Thanks, outdated comment.

File: frontend/src/tests/lib/utils/sns.utils.spec.ts (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/utils/sns.utils.spec.ts:383]: Done

File: frontend/src/tests/lib/components/project-detail/ProjectCommitment.spec.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/project-detail/ProjectCommitment.spec.ts:38]: Done!

File: frontend/src/tests/lib/services/sns-swap-metrics.services.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/services/sns-swap-metrics.services.spec.ts]: done

**Review by lmuntaner — APPROVED** [review]
Perfect, thanks!

**Pull request merged**